src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -581,14 +597,15 @@ protected override TypeSymbol InferTypeOfVarVariable(DiagnosticBag diagnostics)
         }
 
         /// <summary>
-        /// Symbol for an out variable local.
+        /// Symbol for an out variable local that might require type inference during overload resolution, i.e.
+        /// its type is syntactically 'var'.
         /// </summary>
-        private class OutLocalSymbol : SourceLocalSymbol
+        private class PossiblyImplicitlyTypedOutVarLocalSymbol : SourceLocalSymbol
```

Comment:
That would be OK if similar conventions were used for the other local symbol types, but it is not.